docs/DFAnumerics.rst

```diff
@@ -495,53 +495,91 @@ number of particles per kernel radius exceeds a given value :math:`n_{PPK}^{max}
-is a numerical parameter that depends on the mesh size. Its value is set in the configuration (``subgridMixingFactor``) to 100
-by default (ToDo: and should be discussed and further tested).
+is a numerical parameter that depends on the mesh size. Its value is set to 100
+and should be discussed and further tested.
 
 Adding the viscous force
-~~~~~~~~~~~~~~~~~~~~~~~~~
+""""""""""""""""""""""""
 
 The viscous force is added implicitly:
 
 .. math::
   \begin{aligned}
-  \mathbf{F_k^{viscosity}} = &-\frac{1}{2}\rho C_{Lat}\|\mathbf{du}_k^{old}\| A_{Lat}
-  \mathbf{du}_k^{new}\\
-  = &  -\frac{1}{2}\rho C_{Lat}\|\mathbf{u}_k^{old} - \mathbf{\bar{u}}^{old}\| A_{Lat}
-  (\mathbf{u}_k^{new} - \mathbf{\bar{u}}^{old})
+  \mathbf{F_{viscosity}} = &-\frac{1}{2}\rho C_{Lat}\|\mathbf{du}^{old}\| A_{Lat}
+  \mathbf{du}^{new}\\
+  = &  -\frac{1}{2}\rho C_{Lat}\|\mathbf{u}^{old} - \mathbf{\bar{u}}^{old}\| A_{Lat}
+  (\mathbf{u}^{new} - \mathbf{\bar{u}}^{old})
   \end{aligned}
-
 Updating the velocity is done in two steps. First adding the explcit term related to the
 mean mesh velocity and then the implicit term which leads to:
 
 .. math::
-  \mathbf{u}_k^{new} = \frac{\mathbf{u}_k^{old} - C_{vis}\mathbf{\bar{u}}^{old}}{1 + C_{vis}}
+  \mathbf{u}^{new} = \frac{\mathbf{u}^{old} - C_{vis}\mathbf{\bar{u}}^{old}}{1 + C_{vis}}
+
+With :math:`C_{vis} = \frac{1}{2}\rho C_{Lat}\|\mathbf{du}^{old}\| A_{Lat}\frac{dt}{m}`
+
+
+Ata Artificial viscosity
+~~~~~~~~~~~~~~~~~~~~~~~~
 
-With :math:`C_{vis} = \frac{1}{2}\rho C_{Lat}\|\mathbf{du}_k^{old}\| A_{Lat}\frac{dt}{m}`
+An up-winding method based on Lax-Friedrichs scheme
+"""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+Shallow Water Equations are well known for being hyperbolic transport equations.
+They have the particularity of carrying discontinuities or shocks which makes them
+numerically unstable.
+Thus a decentering in time allows to better capture the discontinuity.
+This can be done in the manner of the Lax-Friedrich scheme as described in :cite:`AtSo2005`.
+This decentering in time is formally the same as adding a viscous force.
+Transposed to the SPH method, this viscous force applied on a given particle :math:`k`
+can be expressed as follows:
+
+.. math::
+  \mathbf{F_{{viscosity}_k}} = \sum_{l} \frac{m_l}{\rho_l} \Pi_{kl}
+with :math:`\Pi_{kl} = \lambda_{kl}(\mathbf{u}_l - \mathbf{u}_k) \cdot \mathbf{n}_{kl}`
+
+where :math:`\mathbf{u}_{kl} = \mathbf{u}_k - \mathbf{u}_l` is the relative velocity
+between particle k and l, and :math:`\lambda_{kl} = \frac{c_k+c_l}{2}` with
+:math:`c_k = \sqrt{gh_l}`, the wave speed. The :math:`\lambda_{kl}` is obtained as
+turning expressions related to time and spacial discretization parameters
+into an expression on maximal speed between both particles in the Lax Friedrich
+scheme.
+
+Due to the expression of the viscosity force, it makes sense to
+compute it and add it with the SPH pressure force (for this reason, the
```

Comment:
```suggestion
compute it and add it to the SPH pressure force (for this reason, the
```
??

Reply:
I want to say that we compute the viscosity in the same place as the pressure because we need he loop on neighbor particles. Then we add the viscosity at the same time as we add the pressure.

Reply:
then `and apply it at the same time as the SPH force` ?